.github/workflows/main.yaml

```
@@ -4,38 +4,49 @@ on:
+
       - name: Set up Go
-        uses: actions/setup-go@41dfa10bad2bb2ae585af6ee5bb4d7d973ad74ed # v2
+        uses: actions/setup-go@v2
         with:
           go-version: '1.23' 
 
-      - name: Docker
-        run: make docker-build
+      - name: Extract Git Tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "GIT_TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:44: Double quote to prevent globbing and word splitting

(shellcheck)